File: Projeto_6/src/models/FuncionarioAdministrativo.py
from src.models.UsuarioAutenticavel import UsuarioAutenticavel
from src.models.Relatorio import Relatorio
from src.models.TipoUsuario import TipoUsuario

class FuncionarioAdministrativo(UsuarioAutenticavel, Relatorio):
    def __init__(self, 
                 nome='indefinido', 
                 email='indefinido', 
                 tipo=TipoUsuario.FUNCIONARIO_ADMINISTRATIVO, 
                 senha='', 
                 departamento='indefinido', 
                 cargo='indefinido'):
        # Chamando o construtor da classe UsuarioAutenticavel
        super().__init__(nome, email, tipo, senha)

        # Atributos específicos da classe FuncionarioAdministrativo
        self._departamento = departamento
        self._cargo = cargo

    def autenticar(self, senha):
        # Método de autenticação
        return self._senha == senha

    def gerarRelatorio(self):
        # Método para gerar relatório
        return (
            f"Nome: {self._nome}\n"
            f"Departamento: {self._departamento}\n"
            f"Cargo: {self._cargo}"
        )

File: Projeto_7/Projeto_7- Alissa Garcia/src/FuncionarioAdministrativo.cpp
#include "FuncionarioAdministrativo.h"

FuncionarioAdministrativo::FuncionarioAdministrativo() : UsuarioAutenticavel() {
    departamento = "NULL";
    cargo = "NULL";
}

FuncionarioAdministrativo::FuncionarioAdministrativo(string nome, string email, string senha, string departamento, string cargo)
    : UsuarioAutenticavel(nome, email, TipoUsuario::FUNCIONARIO_ADMINISTRATIVO, senha) {
    this->departamento = departamento;
    this->cargo = cargo;
}

bool FuncionarioAdministrativo::autenticar(string senha) {
    return this->senha == senha;
}

void FuncionarioAdministrativo::gerarRelatorio() const {
    cout << "\n--- RELATÓRIO FUNCIONÁRIO ---\n";
    cout << "Nome: " << nome << endl;
    cout << "Departamento: " << departamento << endl;
    cout << "Cargo: " << cargo << endl;
}
